realm/realm-library/src/main/java/io/realm/internal/Collection.java

```diff
@@ -367,7 +367,7 @@ public Collection(SharedRealm sharedRealm, LinkView linkView, SortDescriptor sor
 
         this.sharedRealm = sharedRealm;
         this.context = sharedRealm.context;
-        this.table = linkView.getTable();
+        this.table = linkView.getTargetTable();
```

Comment:
Yes, it is. it is used to construct the `RealmList.collection`, the `RealmList.collection` is only used for create/remove listeners. listeners won't access the target table, so the bug hasn't been exposed from there.